memmap.go

```diff
@@ -142,13 +142,18 @@ func (m *MemMapFs) Mkdir(name string, perm os.FileMode) error {
 	}
 
 	m.mu.Lock()
+	// double check before create new dir
+	if _, ok := m.getData()[name]; ok {
+		m.mu.Unlock()
+		return &os.PathError{Op: "mkdir", Path: name, Err: ErrFileExists}
+	}
```

Comment:
I would `defer` the `unlock` line 145 so that it happens no matter the path of the code chosen and there is no risk of stale locks
```suggestion
      defer m.mu.Unlock()
	// double check before create new dir
	if _, ok := m.getData()[name]; ok {
		return &os.PathError{Op: "mkdir", Path: name, Err: ErrFileExists}
	}
```

Reply:
LGTM.